### DISCUSSION TITLE
Is there a way to order symbol layers?

### DISCUSSION THREAD

**Discussion by lbernable** [OpenOrienteering/mapper#2051] [Questions & Answers] [answered]
Hello dear Mapper community

First of all, as a new OOM users, as an orienteering fan and as a Free Software enthusiast and advocate, I would like to gratefully thank you for this excellent software, thanks to which I'm currently mapping my city for an urban sprint -- and this works great, and moreover it works perfectly on my Debian daily desktop (You-hou !)

I have a few questions, starting by this one, as I couldn't find the information in the documentation.

When drawing different objects, it seems there is a layered order of them. 

For instance, the 520.1 goes behind the 521. The 501.2 also goes behind the 521.

But on the same map, the order between 520.1 and 501.2 seems to differ. When I draw the 520 on a wide zone, and I want to draw the 501 on top of it, sometimes it works, sometimes the 520 remains on top.

Another use-case : I wanted to draw a 3m-wide high traffic road (dark orange). But this width is only availabile, on the symbol set I have, on a light orange (light traffic road) color code (501.12). So I draw the light orange road, and I wanted to add a 501.21 dark orange on-top, but this layer remains underneath the light orange road itself.

I hope my explanations are clear ^_^ and my question is: is there a possibility to alter on the fly (for specific objects, or eventually for all objects related to a specific code) the order of the printing of the objects?

Many thanks for the answer to this first question. Other questions might follow on other posts ;o)

Best regards,

Loïc

**Comment by mlerjen** [comment]
@lbernable  the order of the "layers" is defined by the order of the colors in the color table. Thus an symbol with one color will always be covered by another symbol with a higher ranked color... and the answer therfore is no, the color order can not be changed locally. (and it is not wise to change the color order in general).
... I ll come back later with specific solutions..

**Reply by mlerjen** [reply]
It is hard to reconstruct the issue as it seems you are not working with the official ISSprOM Symbol Set provided by Mapper. e.g. 501.21 is not part of the related in the Mapper Symbol Set and e.g. does the Mapper Symbol Set indeed provide a line symbol for 3m paved area with heavy traffic (501.18). 
Generally instead of changing the color order you would  avoid to cover the intended area symbol by an other using colors of a higher order. Useful in this context is the "cut away" tool...

**Reply by lbernable** [reply]
Hello mlerjen,

Thanks to your previous answer, which helps me understand the underlying mechanisms of OOM.

The previous set of symbol were derivated from a map initially auto-created by one of the latest versions of OCAD.

Based on your previous comments, I switched to the built-in OOM ISSprOM symbol set. I mapped the best I could the different objects, fine. 

However, I have a similar behaviour, which is even more annoying than before. 

When designing a city road, I tend to draw the road part, with different widths, using symbols such as 501.8 from OOM Symbol Set. I then design the boundaries of private properties, initially using 520.1, but I have been forced to switch it to 501.1 (which can still be fine for me).

But the problem is that at the end, I usually use 501.5 to fill with the right "paved area" color the sidewalk. But problem : this 501.5, even though it uses the same color of the road 501.8, hides the road itself !

Designing "nice" roads, nice boundaries between the sidewalks and the roadway itself, can be very annoying and complex to achieve if I need to create them by hand, without using the predefined 501.8 (and others of the kind), especially for turning roads.

Now the question remains present : how can I achieve such a goal efficiently ?

Attached is a sample of what I want to achieve. I'd like to fill the white areas with "paved area - light traffic" color.

![Capture d’écran du 2022-04-16 00-05-36](https://user-images.githubusercontent.com/81242199/163645720-c2326000-160e-487a-88ce-1d3b12bb5e56.png)

Many thanks for your help,

Loïc

**Reply by lbernable** [reply]
And here is what I get when I try to fill in the white area with 501.5 : 

![Capture d’écran du 2022-04-16 00-13-16](https://user-images.githubusercontent.com/81242199/163648646-36194ab8-bdba-473e-8ee7-ff0ca757b231.png)

You can see the "hand made" "edge of paved area" still appear, but the roads (paved areas 501.7, 501.8, 501.9 etc.) are hidden by the new 501.5.

As long as possible, I would prefer using the standard ISSprOM Symbol Set from OOM rather than using the one initially created by OCAD.

Thanks for your help,

Loïc

**Reply by krticka** [reply]
It is caused a way how you are drawing. Pre-defined roads are never perfect because of colour order which allows easy drawing of junctions, but on the other hand there are cave eats like this one. I draw almost all the roads with separate side lines and than fill it with paved area. It has also advantage, that contours are under side lines (this is especially useful in steep terrains).
BTW: There should be no borders between buildings which are touching, you should merge them.

**Reply by lbernable** [reply]
Hello krticka,

However, this use case works fine with the symbol set imported from OCAD... Is it before the "black" of the predefined roads is underneath the orange of the paved area ? But the black for the "paved area border" is still visible on top of the paved area... Would the fact of using another "black" in the predefined road solve the problem ?

Cheers and thanks for your help -- I don't imagine drawing all the roads uniquely by hand, without using the dedicated predefined roads symbols...

Loïc

**Reply by mlerjen** [reply]
Hi Loïc,
What I usually do in complex situations is to use the predefined roads as helper symbols. (see screencast linked below)

1) Drawing the parts of parallel edges with a predefined width symbol.
2) Drawing the final edge snapping to the outline of the helper.
3) Fixing details at the corners/crossings
4) Filling with paved area.
5) Removing the helper

https://youtu.be/O16skujGfVY

**Reply by lbernable** [reply]
> BTW: There should be no borders between buildings which are touching, you should merge them.

Indeed, but I do it only at the end. Unitary buildings boundaries usually help me drawing the map details, often serving as landmarks when comparing side to side my map and the Google Maps view.

Once the area is drawn on my side, I join the buildings with the "area merge" tool.

Loïc

**Comment by lbernable** [comment, marked as answer]
Hello all

Thanks for your tips and tricks, krticka and mlerjen.

Just for sharing the solution I decided to use : as I didn't want to spend time into redrawing all the roads I already had drawned (as shown in mlerjen video), I created a new "brown 30" color, named "brown 30 under all", and put it the last color in the symbol color list. Then I assigned this color to the "501.5 Paved Area". 

Doing so, I found the same behaviour I had previously, without touching any of the "predefined roads".

Thanks all for your help,

Loïc
